sprint3/zod.js

```diff
@@ -0,0 +1,49 @@
+import { z } from "zod";
+import isUuid from "is-uuid"
+
+const uuidSchema = z.string().refine(isUuid, {
+  message: "Invalid UUID format",
+});
+
+const CreateProduct = z.object({
+  id: uuidSchema,
+  name: z.string().min(2).max(100),
+  description: z.string().max(500).optional(),
+  price: z.number().min(0),
+  tags: z.array(z.string().min(2).max(100)),
+  createdAt: z.date(),
+  updatedAt: z.date(),
```

Comment:
만약, JS의 Date타입이 아니라 String Format의 Date가 들어온다면 (ex: `2025-01-01`) 
`"expected date, received string"` 에러가 발생할 거에요.
JS Date만 받고 싶다면 상관없지만, String 타입의 인자도 허용할 것이라면 아래처럼 작성해주시면 됩니다.

```suggestion
  createdAt: z.coerce.date(),
  updatedAt: z.coerce.date(),
```